### PR TITLE
add salesforce middleware to function builder

### PR DESCRIPTION
**Pull request by jqian2017**
@jkutner , @elbandito , Git repo location of the salesforce middleware build pack is [here](https://github.com/forcedotcom/sf-fx-middleware)

### CODE REVIEW

File: functions-builder.toml [hunk 1]
```diff
@@ -36,6 +40,10 @@ version = "0.4.0"
     id = "heroku/nodejs-npm-buildpack"
     version = "0.1.2"
 
+  [[order.group]]
+    id = "salesforce/sf-fx-middleware"
+    version = "0.0.4"
```
Comment (jkutner) [inline functions-builder.toml]: do we want this to be optional like [procfile buildpack](https://github.com/heroku/pack-images/blob/master/builder.toml#L111):

```
optional = true
```

Comment (jqian2017) [reply functions-builder.toml]: option=true... does it mean if detect fails (return false)... the builder will just build the source w/o the middleware buildpack?

Comment (jkutner) [reply functions-builder.toml]: yes

Comment (elbandito) [reply functions-builder.toml]: @jkutner @jqian2017  At first, I was thinking of having an entirely new `[[order]]` for the middleware + npm + nodejs buildpacks, but having `optional = true` seems like a good approach.  I image down the road, if we have other middleware buildpacks that are mutually exclusive from other buildpacks, we'd want to separate those into separate `[[order]]` sections?

**Pull request merged**